File: id_REVC/PR/python/id_REVC.py
#!/usr/bin/env python
# coding: utf-8

# # Complementing a Strand of DNA
#
# More detailed introduction: http://rosalind.info/problems/revc/
#

filename_in = "rosalind_revc.txt"
filename_out = "rosalind_revc_output.txt"

with open(filename_in, "r") as infile:
    input_string = infile.read()

print("input string:", input_string)
trans_tab = str.maketrans("ATGC", "TACG")  # str.maketrans(map_in, map_out)
reverse_complement = input_string.translate(trans_tab)[::-1].strip() # fix by stripping from string newline!

print("reverse complement:", reverse_complement)

with open(filename_out, "w") as outfile:
    outfile.write(reverse_complement)

File: id_REVC/PR/julia/id_REVC.jl
filename_in = "rosalind_revc.txt"
filename_out = "rosalind_revc_output.txt"

#=
input_string = open(filename_in) do file
    read(file, String)
end
=#

#=
https://stackoverflow.com/questions/35782439/how-to-replace-several-characters-in-a-string-using-julia
https://stackoverflow.com/questions/41106546/julia-trim-string-whitespace-and-check-length
https://en.wikibooks.org/wiki/Introducing_Julia/Strings_and_characters

=#

input_string = strip(read(filename_in, String))

println("Input string:\n", input_string)

transtab = Dict('A' => 'T', 'C' => 'G', 'G' => 'C', 'T' => 'A')

output_string = reverse(join([transtab[c] for c in input_string]))

println("Output string:\n", output_string)
# writing to file
open(filename_out, "w") do file
    write(file, output_string)
end
